platform/ios/src/MGLMapView.mm

```diff
@@ -1128,9 +1128,68 @@ - (void)wakeGL:(__unused NSNotification *)notification
         [self validateLocationServices];
 
         [MGLMapboxEvents pushEvent:MGLEventTypeMapLoad withAttributes:@{}];
+        
+        [self updateLocalization];
+    }
+}
+
+- (void)updateLocalization
+{
+    NSString *mapboxLanguage = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLMapboxLanguage"];
+    NSString *language = [[[NSBundle mainBundle] preferredLocalizations] firstObject];
+    NSString *preferredLanguageCode = [self preferredLanguageCode];
+    
+    if (![mapboxLanguage isEqualToString:language] && ![mapboxLanguage isEqualToString:preferredLanguageCode]) {
+        mapboxLanguage = preferredLanguageCode;
+        [[NSUserDefaults standardUserDefaults] setObject:mapboxLanguage forKey:@"MGLMapboxLanguage"];
+        [self setLabelLanguage:mapboxLanguage];
+        [[NSNotificationCenter defaultCenter] postNotificationName:NSCurrentLocaleDidChangeNotification object:nil];
+    }
+}
+
+- (void)setLabelLanguage:(NSString *)language
+{
+    NSString *preferredNameToken = [NSString stringWithFormat:@"{name_%@}", language];
+    
+    for (MGLSymbolStyleLayer *layer in [[self style] layers]) {
+        if (![layer isKindOfClass:[MGLSymbolStyleLayer class]]) {
+            continue;
+        }
+        
+        if ([layer.text isKindOfClass:[MGLConstantStyleValue class]]) {
```

Comment:
We should only attempt to modify layers that draw from the Mapbox Streets source. A developer should be able to add a shape or vector source using their own custom data with a `name` field and not have MGLMapView attempt to “localize” it to a nonexistent `name_fr` field.